sdk/notificationhubs/notification-hubs/src/serializers/notificationHubJobSerializer.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getDateOrUndefined, getInteger, getString, getStringOrUndefined } from "../utils/xmlUtils";
+
+export enum NotificationHubJobType {
+  /**
+   * Job type to bulk get registrations.
+   */
+  ExportRegistrations = 0,
+
+  /**
+   * Job type to bulk create registrations.
+   */
+  ImportCreateRegistrations = 1,
+
+  /**
+   * Job type to bulk update registrations.
+   */
+  ImportUpdateRegistrations = 2,
+
+  /**
+   * Job type to bulk delete registrations.
+   */
+  ImportDeleteRegistrations = 3,
+
+  /**
+   * Job type to bulk upsert registrations.
+   */
+  ImportUpsertRegistrations = 4
+}
+
+export enum NotificationHubJobStatus {
+  /**
+   * Indicates that the NotificationHubJob was accepted.
+   */
+  Started = 0,
+
+  /**
+   * Indicates that the NotificationHubJob is currently running. Depending on the amount of data, 
+   * a job may stay in this state for several hours.
+   */
+  Running = 1,
+
+  /**
+   * Indicates that the NotificationHubJob was completed successfully. Any output
+   * will be ready where configured via the NotificationHubJob object.
+  */
+  Completed = 2,
+
+  /**
+   * Indicates that the NotificationHubJob has failed.
+  */
+  Failed = 3
+}
+
+/**
+ * Represents a Notification Hub Job.
+ */
+export interface NotificationHubJob {
+  /**
+   * The unique job identifier.
+   */
+  jobId: string;
+
+  /**
+   * The output file name.
+   */
+  outputFileName?: string;
+
+  /**
+   * The file name for the job failures.
+   */
+  failuresFileName?: string;
+
+  /**
+   * The progress for the job.
+   */
+  progress?: number;
+
+  /**
+   * The type of job.
+   */
+  jobType: NotificationHubJobType,
+
+  /**
+   * The status of the job.
+   */
+  status: NotificationHubJobStatus,
+
+  /**
+   * The output container URL.
+   */
+  outputContainerUrl: string;
+
+  /**
+   * The import file URL.
+   */
+  importFileUrl?: string;
+
+  /**
+   * The input properties for the notification hub job.
+   */
+  inputProperties?: Record<string, string>;
+
+  /**
+   * Gets the notification hub job failure message.
+   */
+  failure?: string;
```

Comment:
In core-lro, a similar property is called [error](https://github.com/Azure/azure-sdk-for-js/blob/2bb81e90e7d4652cd6ce39ee491722e6c8e93164/sdk/core/core-lro/src/pollOperation.ts#L33), perhaps it is a good idea to align the naming?

Reply:
This is a direct name from the Track 1 library